modules/mmf/Pokemon.py

```diff
@@ -0,0 +1,211 @@
+}
+
+PokemonValidator = fastjsonschema.compile(pokemon_schema) # Validate the data from the mmf, sometimes it sends junk
+
+Natures = [
+    "Hardy",
+    "Lonely",
+    "Brave",
+    "Adamant",
+    "Naughty",
+    "Bold",
+    "Docile",
+    "Relaxed",
+    "Impish",
+    "Lax",
+    "Timid",
+    "Hasty",
+    "Serious",
+    "Jolly",
+    "Naive",
+    "Modest",
+    "Mild",
+    "Quiet",
+    "Bashful",
+    "Rash",
+    "Calm",
+    "Gentle",
+    "Sassy",
+    "Careful",
+    "Quirky"
+]
+
+HiddenPowers = [
+    "Fighting",
+    "Flying",
+    "Poison",
+    "Ground",
+    "Rock",
+    "Bug",
+    "Ghost",
+    "Steel",
+    "Fire",
+    "Water",
+    "Grass",
+    "Electric",
+    "Psychic",
+    "Ice",
+    "Dragon",
+    "Dark"
+]
+
+def EnrichMonData(pokemon: dict): # Function to add information to the pokemon data extracted from Bizhawk
+    try:
+        if pokemon["speciesName"].isalpha():
+            trainer = GetTrainer()
+            pokemon["name"] = pokemon["speciesName"].capitalize() # Capitalise name
+            pokemon["metLocationName"] = location_list[pokemon["metLocation"]] # Add a human readable location
+            pokemon["type"] = pokemon_list[pokemon["name"]]["type"] # Get pokemon types
+            pokemon["ability"] = pokemon_list[pokemon["name"]]["ability"][0] # Get pokemon abilities
+            pokemon["hiddenPowerType"] = HiddenPowers[int(numpy.floor((((pokemon["hpIV"] % 2)+(2*(pokemon["attackIV"] % 2))+(4*(pokemon["defenseIV"] % 2))+(8*(pokemon["speedIV"] % 2))+(16*(pokemon["spAttackIV"] % 2))+(32*(pokemon["spDefenseIV"] % 2)))*15)/63))]
+            pokemon["nature"] = Natures[pokemon["personality"] % 25] # Get pokemon nature
+            pokemon["zeroPadNumber"] = f"{pokemon_list[pokemon['name']]['number']:03}" # Get zero pad number - e.g.: #5 becomes #005
+            pokemon["itemName"] = item_list[pokemon["heldItem"]] # Get held item's name
+
+            pBin = format(pokemon["personality"], "032b") # Convert personality ID to binary
+            pokemon["shinyValue"] = int(bin(int(pBin[:16], 2) ^ int(pBin[16:], 2) ^ trainer["tid"] ^ trainer["sid"])[2:], 2) # https://bulbapedia.bulbagarden.net/wiki/Personality_value#Shininess
+            pokemon["shiny"] = True if pokemon["shinyValue"] < 8 else False
+
+            pokemon["move_1"] = pokemon["moves"][0]
+            pokemon["move_2"] = pokemon["moves"][1]
+            pokemon["move_3"] = pokemon["moves"][2]
+            pokemon["move_4"] = pokemon["moves"][3]
+            pokemon["move_1_pp"] = pokemon["pp"][0]
+            pokemon["move_2_pp"] = pokemon["pp"][1]
+            pokemon["move_3_pp"] = pokemon["pp"][2]
+            pokemon["move_4_pp"] = pokemon["pp"][3]
+            pokemon["type_1"] = pokemon["type"][0]
+
+            pokemon["type_2"] = "None"
+            if len(pokemon["type"]) == 2:
+                pokemon["type_2"] = pokemon["type"][1]
+
+            pokemon["enrichedMoves"] = []
+            for move in pokemon["moves"]:
```

Comment:
can use list comprehension
'''
pokemon["enrichedMoves"]=[move for move in pokemon["moves"]]
'''